tests/test_datacube_axes_init.py

```diff
@@ -9,6 +12,24 @@
 
 class TestInitDatacubeAxes:
     def setup_method(self, method):
+        nexus_url = "https://get.ecmwf.int/test-data/polytope/test-data/foo.grib"
+
+        local_directory = "./tests/data"
+
+        if not os.path.exists(local_directory):
+            os.makedirs(local_directory)
+
+        # Construct the full path for the local file
+        local_file_path = os.path.join(local_directory, "foo.grib")
+
+        if not os.path.exists(local_file_path):
+            session = requests.Session()
+            response = session.get(nexus_url)
```

Comment:
In general it's frowned upon to have unit tests which require an internet connection. IMO you should label all tests which are downloading data, then developers can easily choose to skip or include the tests which require an internet connection depending on their situation.